src/test/java/StringManipulationTest.java

```diff
@@ -88,27 +129,60 @@ public void testGeSubStrings1() {
     }
 
     @Test
+    // Checks behavior for start or end word = 0
     public void testGeSubStrings2() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("I <3 strings");
+        Executable executable = new Executable() {
+            public void execute() {
+                manipulatedstring.getSubStrings(0, 0);
+            }
+        };
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, executable);
+        assertEquals("startWord must be greater than endWord. Both startWord and endWord must be greater than 0.", exception.getMessage());
     }
     @Test
+    // Checks behavior for start word > end word
     public void testGeSubStrings3() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("I <3 strings");
+        Executable executable = new Executable() {
+            public void execute() {
+                manipulatedstring.getSubStrings(3, 2);
+            }
+        };
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, executable);
+        assertEquals("startWord must be greater than endWord. Both startWord and endWord must be greater than 0.", exception.getMessage());
     }
     @Test
+    // Checks behavior for end word > than string length
     public void testGeSubStrings4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("I <3 strings");
+        Executable executable = new Executable() {
+            public void execute() {
+                manipulatedstring.getSubStrings(1, 5);
+            }
+        };
+        IndexOutOfBoundsException exception = assertThrows(IndexOutOfBoundsException.class, executable);
+        assertEquals("endWord cannot be greater than length of substr", exception.getMessage());
     }
     @Test
+    // Checks behavior for start word and end word being equal
     public void testGeSubStrings5() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("I <3 strings");
+        String [] sStings = manipulatedstring.getSubStrings(1, 1);
+
+        assertEquals(sStings[0], "I");
     }
     @Test
+    // Extra test?
     public void testGeSubStrings6() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("I am not sure why there is a 6th test here");
+        String [] sStings = manipulatedstring.getSubStrings(1, 1);
+
+        assertEquals(sStings[0], "I");
     }
 
     @Test
+    // Checks behavior for restoring string
```

Comment:
//for this test I would try something like this
public void testRestoreString1() {
    manipulatedstring.setString("art");
    manipulatedstring.reverseString();
    manipulatedstring.removeNthCharacter(1, true);
    manipulatedstring.restoreString();
    String restoredString = manipulatedstring.getString();
    assertEquals("Expected restored string", "art", restoredString);
}